### DISCUSSION TITLE
How does lazy load works in packer.nvim, why "module not found"?

### DISCUSSION THREAD

**Discussion by winkee01** [wbthomason/packer.nvim#896] [Q&A]
When I put a plugin in **lazy load mode**, `nvim` complains that **module not found**, because packer.nvim downloaded it into `.../pack/packer/opt/` directory but didn't set the plugin's path into runtimepath, why?

![image](https://user-images.githubusercontent.com/49930785/165964339-8ea756f2-c0d6-4507-afa9-c957a2677944.png)

```
use({
  'L3MON4D3/LuaSnip', 
  requires = {'rafamadriz/friendly-snippets'}, 
  after = 'cmp_luasnip'
})
```

How can I fix it? (or how to tell nvim to set the runtimepath?)

**Comment by milanglacier** [comment]
oneshot solution, call `vim.cmd [[packadd LuaSnip]]` at your config file of `nvim cmp`.
If you don't spend useless time on figuring out the loading dependency, this is the easist way

**Reply by winkee01** [reply]
@milanglacier I do want to know how the `opt/` dir works because I keep encountering such problems

**Reply by milanglacier** [reply]
You can take a look at compiled.lua generated by packer. In short, it just defines that when certain condition is satisfied, the vim.cmd [[packadd xxx]] will be called and then your config will be called.
`Packadd` command will add the the 1st level sub folder in opt folder ( vim will search for the folder name) into its runtime path. That’s it, nothing special.

**Reply by winkee01** [reply]
Great explanations, thanks a lot!
